packages/streaming-image-volume-loader/src/StreamingImageVolume.ts

```diff
@@ -434,7 +435,30 @@ export default class StreamingImageVolume extends ImageVolume {
       // when we load directly into the Volume cache
       function callLoadImage(imageId, imageIdIndex, options) {
         return imageLoader.loadImage(imageId, options).then(
-          () => {
+          (image) => {
+            const scalarData = this.scalarData;
+            if (scalarData.buffer instanceof ArrayBuffer) {
+              const offset = options.targetBuffer.offset; // in bytes
+              const length = options.targetBuffer.length; // in frames
+              try {
+                if (scalarData instanceof Float32Array) {
+                  const floatView = new Float32Array(image.pixelData);
+                  if (floatView.length !== length) {
+                    throw 'Error pixelData length does not match frame length';
+                  }
+                  scalarData.set(floatView, offset / 4);
+                }
+                if (scalarData instanceof Uint8Array) {
+                  const intView = new Uint8Array(image.pixelData);
+                  if (intView.length !== length) {
+                    throw 'Error pixelData length does not match frame length';
+                  }
+                  scalarData.set(intView, offset);
+                }
+              } catch (e) {
+                console.error(e);
+              }
+            }
```

Comment:
These all better be refactor into its own function e.g., `handleArrayBufferLoad` or somthing similar